File: apps/sdkwork-appstore-mini-program/src/bootstrap/iamRuntime.ts
/**
 * Appbase IAM runtime wiring for the mini program root.
 *
 * Authority: `APP_SDK_INTEGRATION_SPEC.md` and
 * `IAM_LOGIN_INTEGRATION_SPEC.md`. The root owns exactly one TokenManager and
 * one IAM runtime; feature packages receive session state by injection.
 */

export interface MiniProgramIamSession {
  accessToken?: string;
  refreshToken?: string;
}

let session: MiniProgramIamSession = {};

export function getIamSession(): MiniProgramIamSession {
  return session;
}

export function setIamSession(next: MiniProgramIamSession): void {
  session = next;
}

export function clearIamSession(): void {
  session = {};
}

File: apps/sdkwork-appstore-flutter-mobile/lib/bootstrap/iam_runtime.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

/// Appbase IAM runtime wiring for the Flutter root.
///
/// Authority: `APP_SDK_INTEGRATION_SPEC.md` and
/// `IAM_LOGIN_INTEGRATION_SPEC.md`. The root owns exactly one token manager
/// and one IAM runtime; feature packages receive session state by injection.
AppstoreIamRuntime? _runtime;

AppstoreIamRuntime createIamRuntime() {
  _runtime ??= AppstoreIamRuntime();
  return _runtime!;
}

AppstoreIamRuntime getIamRuntime() => _runtime ?? createIamRuntime();
